src/psyclone/tests/psyir/nodes/reference_test.py

```diff
@@ -194,3 +195,160 @@ def test_array_create_invalid3():
+
+    # Index out of bounds
+    array_reference = Array.create(symbol,[one])
+    with pytest.raises(ValueError) as excinfo:
+        array_reference.is_full_range(1)
+    assert ("In Array 'my_array' the specified index '1' must be less than "
+            "the number of dimensions '1'." in str(excinfo.value))
+
+    # Array dimension is not a Range
+    assert not array_reference.is_full_range(0)
+
+    # Check LBOUND
+    # Array dimension range lower bound is not a binary operation
+    my_range = Range.create(one, one, one)
+    array_reference = Array.create(symbol,[my_range])
+    assert not array_reference.is_full_range(0)
+
+    # Array dimension range lower bound is not an LBOUND binary operation
+    my_range = Range.create(ubound, one, one)
+    array_reference = Array.create(symbol,[my_range])
+    assert not array_reference.is_full_range(0)
+
+    # Array dimension range lower bound is an LBOUND binary operation
+    # with the first value not being a reference
+    lbound_error = BinaryOperation.create(BinaryOperation.Operator.LBOUND,
+                                          zero, zero)
+    my_range = Range.create(lbound_error, one, one)
+    array_reference = Array.create(symbol,[my_range])
+    assert not array_reference.is_full_range(0)
+
+    # Array dimension range lower bound is an LBOUND binary operation
+    # with the first value being a reference to a different symbol
+    lbound_error = BinaryOperation.create(BinaryOperation.Operator.LBOUND,
+                                          reference_error, zero)
+    my_range = Range.create(lbound_error, one, one)
+    array_reference = Array.create(symbol,[my_range])
+    assert not array_reference.is_full_range(0)
+
+    # Array dimension range lower bound is an LBOUND binary operation
+    # with the second value not being a literal.
+    lbound_error = BinaryOperation.create(BinaryOperation.Operator.LBOUND,
+                                          reference, reference)
+    my_range = Range.create(lbound_error, one, one)
+    array_reference = Array.create(symbol,[my_range])
+    assert not array_reference.is_full_range(0)
+
+    # Array dimension range lower bound is an LBOUND binary operation
+    # with the second value not being an integer literal.
+    lbound_error = BinaryOperation.create(
+        BinaryOperation.Operator.LBOUND, reference,
+        Literal("1.0", DataType.REAL))
+    my_range = Range.create(lbound_error, one, one)
+    array_reference = Array.create(symbol,[my_range])
+    assert not array_reference.is_full_range(0)
+
+    # Array dimension range lower bound is an LBOUND binary operation
+    # with the second value being an integer literal with the wrong
+    # value (should be 0 as this dimension index is 0).
+    lbound_error = BinaryOperation.create(
+        BinaryOperation.Operator.LBOUND, reference, one)
+    my_range = Range.create(lbound_error, one, one)
+    array_reference = Array.create(symbol,[my_range])
```

Comment:
A lot of your `create()` calls are missing whitespace after the comma.

Reply:
Oops. Updated.